app/main.py

```diff
@@ -2,0 +1,70 @@
+class Animal:
+    alive = []  # Class attribute to track all alive animals
+
+    def __init__(self, name):
+        self.name = name
+        self.health = 100
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __del__(self):
+        Animal.alive.remove(self)  # Remove the animal from the alive list when it dies
+
+    @classmethod
+    def remove_dead(cls, animal):
+        """Remove dead animals from the alive list."""
+        if animal.health <= 0:
+            cls.alive.remove(animal)
+
+    def __str__(self):
+        """Return a custom string for Animal's state."""
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self):
+        """Toggle the 'hidden' state."""
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore):
+        """Carnivores can bite herbivores, reducing their health."""
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                herbivore.__del__()  # Remove from alive if health drops to 0
```

Comment:
Directly calling `__del__` is not recommended. Instead, use the `remove_dead` method to handle the removal of dead animals from the `alive` list.